librz/include/rz_search.h

```diff
@@ -12,117 +12,46 @@ extern "C" {
 
 RZ_LIB_VERSION_HEADER(rz_search);
 
-enum {
-	RZ_SEARCH_ESIL,
-	RZ_SEARCH_KEYWORD,
-	RZ_SEARCH_REGEXP,
-	RZ_SEARCH_PATTERN,
-	RZ_SEARCH_STRING,
-	RZ_SEARCH_XREFS,
-	RZ_SEARCH_AES,
-	RZ_SEARCH_PRIV_KEY,
-	RZ_SEARCH_DELTAKEY,
-	RZ_SEARCH_MAGIC,
-	RZ_SEARCH_LAST
-};
-
-#define RZ_SEARCH_DISTANCE_MAX 10
-
-#define RZ_SEARCH_KEYWORD_TYPE_BINARY 'i'
-#define RZ_SEARCH_KEYWORD_TYPE_STRING 's'
-
-typedef struct rz_search_keyword_t {
-	ut8 *bin_keyword;
-	ut8 *bin_binmask;
-	ut32 keyword_length;
-	ut32 binmask_length;
-	void *data;
-	int count;
-	int kwidx;
-	int icase; // ignore case
-	int type;
-	ut64 last; // last hit hint
-} RzSearchKeyword;
+typedef struct rz_search_opt_t RzSearchOpt;
+
+typedef struct rz_search_collection_t RzSearchCollection;
 
 typedef struct rz_search_hit_t {
-	RzSearchKeyword *kw;
-	ut64 addr;
+	char *metadata; ///< Metadata for extra details (can be NULL)
```

Comment:
This seems to save mostly the type of the search hit or the things to search for (I saw `aes` and `regex`). Maybe changing it to an enum (`RzSearchMetaDataType type`) and a add a `void *meta_data` to save more complex stuff. This way we can save allocation of simple strings.

Reply:
I added it for the flags so we know it's hit.regex or hit.rsa etc...

Reply:
sure for me.